### ISSUE TITLE
Migration tool applies assertion intent too broadly in "Pop Pop Win"

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/sdk#40700]
The tool migration for pop pop win produces the following code:

```dart
ResourceManager _resourceManager;

void initialize(ResourceManager resourceManager) {
  /* if (_resourceManager != null) */ throw StateError("already initialized.");
}
```

This is clearly wrong.

The reason it does this is an assertion in a later function that the resource provider has been initialized:

```dart
void _playAudio(String name) {
  assert(_resourceManager != null, 'Not initialized');
  ...
}
```

https://github.com/dart-lang/sample-pop_pop_win/blob/master/lib/src/audio.dart#L39

A simple solution would be to do postdominator analysis, so that the intent is not propagated in this case.

However, is noteworthy that the human migration is to make this late, and non-nullable. This leaves an interesting question that if we do that, how would the tool know that this is a non-null variable *and also* that the *true* branch of the non-null check should be removed??

**Comment by stereotype441** [comment]
Upon further investigation, the problem is actually that when analyzing this line:
```dart
ResourceManager _resourceManager;`
```
the migration tool fails to account for the fact ath `_resourceManager` is implicitly initialized to `null`.  I'm working on a fix.

**Comment by kevmoo** [comment]
Awesome!

**Comment by stereotype441** [comment]
Fixed by 85f6d51c3fd10b8a51690f5124e312269525c79a.
